rectools/models/rank.py

```diff
@@ -191,15 +203,50 @@ def rank(
 
         real_k = min(k, object_factors.shape[0])
 
-        ids, scores = implicit.cpu.topk.topk(  # pylint: disable=c-extension-no-member
-            items=object_factors,
-            query=subject_factors,
-            k=real_k,
-            item_norms=object_norms,  # query norms for COSINE distance are applied afterwards
-            filter_query_items=filter_query_items,  # queries x objects csr matrix for getting neginf scores
-            filter_items=None,  # rectools doesn't support blacklist for now
-            num_threads=num_threads,
-        )
+        if use_gpu and not HAS_CUDA:
+            warnings.warn("Forced rank() on CPU")
```

Comment:
If I understand you right, I should keep these lines and remove HAS_CUDA guards in tests?
```py
@pytest.mark.parametrize("use_gpu", (False, True) if HAS_CUDA else (False,))
```

Reply:
Yes, let's just do `@pytest.mark.parametrize("use_gpu", (False, True))`. Because you wrote a fallback behaviour and now we expect same results from cpu and gpu on all tests.

Reply:
Done